react/features/video-layout/middleware.web.js

```diff
@@ -0,0 +1,43 @@
+// @flow
+
+import VideoLayout from '../../../modules/UI/videolayout/VideoLayout.js';
+import UIEvents from '../../../service/UI/UIEvents';
+
+import {
+    DOMINANT_SPEAKER_CHANGED,
+    PIN_PARTICIPANT
+} from '../base/participants';
+import { MiddlewareRegistry } from '../base/redux';
+
+declare var APP: Object;
+
+/**
+ * Middleware which intercepts actions and updates the legacy component
+ * {@code VideoLayout} as needed. The purpose of this middleware is to redux-ify
+ * {@code VideoLayout} without having to simultaneously react-ifying it.
+ *
+ * @param {Store} store - The redux store.
+ * @returns {Function}
+ */
+// eslint-disable-next-line no-unused-vars
+MiddlewareRegistry.register(store => next => action => {
```

Comment:
Are you sure you want to have a web specific middleware? Or it's just not easy to have shared code at this stage? Because base/conference and base/participants handle these events for mobile normally